runtime/fastly/builtins/fetch-event.cpp

```diff
@@ -670,9 +670,10 @@ bool response_promise_then_handler(JSContext *cx, JS::HandleObject event, JS::Ha
 
   bool streaming = false;
   if (Response::is_grip_upgrade(response_obj)) {
+    auto grip_upgrade_request = Response::grip_upgrade_request(response_obj);
```

Comment:
So all this work can actually be simplified since we do have the request handle already available in this function via the Request slot on the fetch event itself:

```
JS::RootedObject request(
      cx, &JS::GetReservedSlot(self, static_cast<uint32_t>(Slots::Request)).toObject());
MOZ_ASSERT(request);
auto req_handle = Request::request_handle(request);
MOZ_ASSERT(req_handle.is_valid());
```

Reply:
Thanks for the comment!

The request handle passed in to `redirect_to_grip_proxy_v2()` doesn't necessarily have to be the request handle attached to the fetch event. This way you can use modified headers and/or url. So I want to save off the request that was passed in as the first parameter of `createFanoutHandoff()` and pass that.

Reply:
Ahh right, sorry I traced the wrong Response::create here!